### PR TITLE
Improve area attack and reclaim orders

### PR DESCRIPTION
**Pull request by clyfordv**
Click and drag to adjust the radius of area reclaim. Only works when in reclaim command mode.

### CODE REVIEW

**Comment by arch625** [comment]
Could you attach a video clip showing the feature?

**Comment by Garanas** [comment]
@arch625 

https://github.com/FAForever/fa/assets/15778155/0f993af6-da10-4185-8510-ceba818765cd

**Comment by Garanas** [comment]
Implementation of [this article](https://stackoverflow.com/questions/28567166/uniformly-distribute-x-points-inside-a-circle) on Stack Overflow:

https://github.com/FAForever/fa/assets/15778155/98c8d7dc-36a0-4512-992a-c05b235950b1

**Comment by Garanas** [comment]
https://github.com/FAForever/fa/assets/15778155/42d692ba-724d-4a9e-8707-681c7a2b0642

**Comment by RegalEagIe** [comment]
Oh dear gog... This is amazing...

**Comment by Garanas** [comment]
https://github.com/FAForever/fa/assets/15778155/c6329112-b356-47ee-b94c-610588106bba

File: lua/ui/controls/worldview.lua [hunk 1]
```diff
@@ -1362,26 +1363,30 @@ WorldView = ClassUI(moho.UIWorldView, Control) {
...
         end
     end,
 
     OnRenderWorld = function (self, delta)
         -- called when custom world rendering is enabled
-        for data, _ in self.UI_DrawShapesTable do
-            if data.shape == 'Circle' then
+        for index, data in self.DrawShapesTable do
+            if data.remove then
+                self.DrawShapesTable[index] = nil
```
Comment (Garanas) [inline lua/ui/controls/worldview.lua:1381]: You can't add or remove elements from a table while iterating over it; it will cause hiccups in the iteration

**Comment by ChessBerry** [comment]
Something changed in between implementations: With the one on the current commit, ea99794ce8bd264d39645eb1ff2cd96cccda327c, it's now seemingly not possible anymore to reclaim individual props w/o also ordering reclaim commands to everything around them.

With the one from a couple of days ago, 0b8523c4e6a1998f705019dffd2ff3a1507ccfea, this was possible.

Comparison, New vs Old:
https://github.com/FAForever/fa/assets/90977374/e75bf300-4667-4dc4-90bd-acd6ba38c4e6

The ability to reclaim individual props w/o issuing an area reclaim command is pretty important for high level play and, even more significantly, it's necessary to get people to play on faf dev more than once.

**Comment by Garanas** [comment]
@ChessBerry great find; it's because I accidentally removed the minimum radius criteria which defaulted to 1. Which is the size of a wall. I'll add that back in

**Pull request merged**